File: LeetCode/1248/main.cpp
class Solution {
public:

    int helper(vector<int>& arr, int k) {

        int n = arr.size();
        int left = 0, ans = 0;

        for(int i = 0; i < n; i++) {
            if(arr[i] % 2 == 1) {
                k--;
            }
            while(k < 0) {
                if(arr[left] % 2 == 1) {
                    k++;
                }
                left++;
            }
            ans += (i - left + 1);
        }

        return ans;
    }

    int numberOfSubarrays(vector<int>& arr, int k) {
        return helper(arr, k) - helper(arr, k - 1);
    }
};

File: LeetCode/1248/main.py
class Solution:
    def numberOfSubarrays(self, arr: List[int], k: int) -> int:

        def atmost(arr, k):
            n = len(arr)
            left, ans = 0, 0

            for i in range(n):

                if (arr[i] % 2 == 1):
                    k -= 1

                while (k < 0):
                    if (arr[left] % 2 == 1):
                        k += 1
                    left += 1

                ans += (i - left + 1)

            return ans

        return atmost(arr, k) - atmost(arr, k-1)
